server/src-lib/Hasura/Prelude.hs

```diff
@@ -142,16 +139,6 @@ spanMaybeM f = go . toList
       Just y  -> first (y:) <$> go xs
       Nothing -> pure ([], l)
 
--- | Efficiently coerce a set from one type to another.
---
--- This has the same safety properties as 'Set.mapMonotonic', and is equivalent
--- to @Set.mapMonotonic coerce@ but is more efficient. This is safe to use when
--- both @a@ and @b@ have automatically derived @Ord@ instances.
---
--- https://stackoverflow.com/q/57963881/176841
-coerceSet :: Coercible a b=> Set.Set a -> Set.Set b
-coerceSet = unsafeCoerce
-
```

Comment:
Not sure why this is removed from here. Is this change relevant to this PR?

Reply:
No, it's not relevant. Would you rather this change be merged separately?

Reply:
I guess it's just dead code removal. It's fine this way

Reply:
We should check that Pro doesn't use it.

Reply:
@nicuveo According to some quick `grep`s, it's not used by pro.

Reply:
Perfect, then. LGTM! :)